openapi.yaml

```
@@ -5660,16 +5665,39 @@ components:
       type: string
       format: date-time
       description: >-
-        Date and time of creation, formatted as a [RFC
-        3339](https://www.rfc-editor.org/rfc/rfc3339.html) date-time.
+        Date and time of creation (for batch jobs: the status 'created' was set),
+        formatted as an [RFC 3339](https://www.rfc-editor.org/rfc/rfc3339.html) date-time.
       example: '2017-01-01T09:32:12Z'
     updated:
       type: string
       format: date-time
       description: >-
-        Date and time of the last status change, formatted as a [RFC
-        3339](https://www.rfc-editor.org/rfc/rfc3339.html) date-time.
+        Date and time of the last status change,
+        formatted as an [RFC 3339](https://www.rfc-editor.org/rfc/rfc3339.html) date-time.
+        If the status is `error`, `canceled` or `finished`,
+        this is the time when the job has ended.
       example: '2017-01-01T09:36:18Z'
+    queued:
+      type: string
+      format: date-time
+      description: >-
+        Date and time of queueing the batch job (i.e., when the status 'queued' was set),
+        formatted as a [RFC 3339](https://www.rfc-editor.org/rfc/rfc3339.html) date-time.
+      example: '2017-01-01T09:34:00Z'
+    started:
+      type: string
+      format: date-time
+      description: >-
+        Date and time when the batch job started processing (i.e., when the status 'running' was set),
+        formatted as a [RFC 3339](https://www.rfc-editor.org/rfc/rfc3339.html) date-time.
+      example: '2017-01-01T09:36:18Z'
+    expires:
+      type: string
+      format: date-time
+      description: >-
+        Time until which the assets are accessible, in UTC. Formatted as
```

```suggestion
        Time until which the assets (e.g., batch job results) are available for download, in UTC. Formatted as
```

---

That might be a bit misleading. 
expires can be used for when the signed URLs expire and also for when the data is deleted, I think. That's why I chose accessible. The addition in brackets is fine though.

```suggestion
        Time until which the assets (e.g., batch job results) are accessible, in UTC. Formatted as
```

---

ok good point. 
But indeed, we should separate the expiry of signed URLs (which can be refreshed easily) and the expiry for download availability in general (which is final and non-recoverable).

I think the general availability of results should be a global property on  `GET /jobs/{job_id}` or `GET /jobs/{job_id}/results`, while the signed URL expiry should be closer to the links/URLs themselves (e.g. different signed URLs within the same resource might have different expiry for some advanced reason)

---

Okay, I've updated the PR accordingly:

- Clarified usage of expires, removed from the batch job info endoint
- Added unpublished to batch job info endpoint and batch job results
- Added all the timestamps also to the Collection-typed result response, previously it was only defined in the Item-typed result response